internal/driver/driver.go

```diff
@@ -0,0 +1,646 @@
+	asyncCh       chan<- *sdkModels.AsyncValues
+	deviceCh      chan<- []sdkModels.DiscoveredDevice
+	activeDevices map[string]*Device
+	rtspHostName  string
+	rtspTcpPort   string
+	mutex         sync.Mutex
+}
+
+// NewProtocolDriver initializes the singleton Driver and returns it to the caller
+func NewProtocolDriver() *Driver {
+	once.Do(func() {
+		driver = new(Driver)
+	})
+	return driver
+}
+
+// Initialize performs protocol-specific initialization for the device service.
+func (d *Driver) Initialize(lc logger.LoggingClient, asyncCh chan<- *sdkModels.AsyncValues,
+	deviceCh chan<- []sdkModels.DiscoveredDevice) error {
+	d.mutex.Lock()
+	defer d.mutex.Unlock()
+
+	d.lc = lc
+	d.asyncCh = asyncCh
+	d.deviceCh = deviceCh
+	d.ds = service.RunningService()
+	d.activeDevices = make(map[string]*Device)
+	d.wg = new(sync.WaitGroup)
+
+	h := NewHttpHandler(d)
+	if err := d.ds.AddRoute(common.ApiBase+ApiRefreshDevicePaths, h.RefreshExistingDevicePaths, http.MethodPost); err != nil {
+		return fmt.Errorf("failed to add API route %s, error: %s", ApiRefreshDevicePaths, err.Error())
+	}
+
+	rtspServerHostName, ok := service.DriverConfigs()[RtspServerHostName]
+	if !ok {
+		rtspServerHostName = DefaultRtspServerHostName
+		d.lc.Warnf("service config %s not found. Use the default value: %s", RtspServerHostName, DefaultRtspServerHostName)
+	}
+	d.lc.Debugf("RTSP server hostname: %s", rtspServerHostName)
+	d.rtspHostName = rtspServerHostName
+
+	rtspPort, ok := service.DriverConfigs()[RtspTcpPort]
+	if !ok {
+		rtspPort = DefaultRtspTcpPort
+		d.lc.Warnf("service config %s not found. Use the default value: %s", RtspTcpPort, DefaultRtspTcpPort)
+	}
+	d.lc.Debugf("RTSP TCP port: %s", rtspPort)
+	d.rtspTcpPort = rtspPort
+
+	d.lc.Info("Initializing cameras...")
+	for _, dev := range d.ds.Devices() {
+		d.lc.Infof("initialize device %s", dev.Name)
+		activeDevice, edgexErr := d.newDevice(dev.Name, dev.Protocols)
+		if edgexErr != nil {
+			d.lc.Error(edgexErr.Error())
+			continue
+		}
+		d.activeDevices[dev.Name] = activeDevice
+	}
+
+	for _, dev := range d.activeDevices {
+		if dev.autoStreaming {
+			edgexErr := d.startStreaming(dev)
+			if edgexErr != nil {
+				d.lc.Errorf("failed to start video streaming for device %s, error: %s", dev.name, edgexErr)
+			}
+		}
+	}
+
+	// Make sure the paths of existing devices are up to date.
+	go d.RefreshExistingDevicePaths()
+
+	return nil
+}
+
+func (d *Driver) HandleReadCommands(deviceName string, protocols map[string]models.ProtocolProperties,
+	reqs []sdkModels.CommandRequest) ([]*sdkModels.CommandValue, error) {
+	d.lc.Debugf("Driver.HandleReadCommands: protocols: %v resource: %v attributes: %v", protocols,
+		reqs[0].DeviceResourceName, reqs[0].Attributes)
+	var err error
+	var responses = make([]*sdkModels.CommandValue, len(reqs))
+
+	device, edgexErr := d.getDevice(deviceName)
+	if edgexErr != nil {
+		return responses, errors.NewCommonEdgeXWrapper(edgexErr)
+	}
+	v4l2Device, err := v4l2device.Open(device.path)
```

Comment:
`v4l2` has no meaning/context here. Use better name
```suggestion
	cameraDevice, err := usbdevice.Open(device.path)
```

Reply:
Modified. https://github.com/edgexfoundry-holding/device-usb-camera/pull/11/commits/b915490962f8488701e51706647b02e88983813a